examples/cpp/hello_world/CLIParser.hpp

```diff
@@ -68,34 +75,49 @@ class CLIParser
             char* argv[])
     {
         hello_world_config config;
-        config.entity = "";
-        config.pub_config.samples = 0;
-        config.sub_config.samples = 0;
-        config.sub_config.use_waitset = false;
 
-        for (int i = 1; i < argc; ++i)
+        std::string first_argument = argv[1];
```

Comment:
Please check argc before accessing. If not arguments are passed, this would be our of bounds. In fact, if no args are passed, we probably should `print_help(EXIT_FAILURE)`